dmutils/email/dm_mailchimp.py

```diff
@@ -28,11 +30,11 @@ def get_email_hash(email_address):
         formatted_email_address = six.text_type(email_address.lower()).encode('utf-8')
         return md5(formatted_email_address).hexdigest()
 
-    def timeout_retry(method):
-        def wrapper(self, *args, **kwargs):
+    def timeout_retry(self, method):
```

Comment:
I thought we had a generic retry wrapper (from an external lib?) which we used elsewhere...?

But yeah it seems it would make sense to have the default number of retries non-zero. It's an odd "default desired behaviour" to "assume" - if they didn't want any retries, why did they apply the wrapper?

Reply:
The number of retries is set to three when we instantiate the Mailchimp client in the script, here: https://github.com/alphagov/digitalmarketplace-scripts/blob/master/scripts/upload-dos-opportunities-email-list.py#L68

I agree it would make sense to set a non-zero default though.

Elsewhere we use backoff to retry stuff. I think the timeout_retry method is used as it has the quirk of only retrying it the caught HTTPError is a 504. It might be possible to set this up with backoff, but I thought I'd try and keep the diff as minimal as possible here.

Reply:
Actually, the more I think about it the more I think having the default of zero makes sense. The wrapper is in there so that it can retry if that's what you want to do. It's an optional extra. So if you just used the client without even thinking about if retries was a thing you want, then you get the expected behaviour. If you then decide, hang on, I'd like to retry this if it fail, the option is there.

Reply:
Yeah.... I guess the difference is that `retries` is a constructor arg to `DMMailChimpClient`, not `timeout_retry` itself

Reply:
`picard_facepalm.png` so it is. I am convinced 🐱